graphitron-codegen-parent/graphitron-java-codegen/src/test/java/no/sikt/graphitron/queries/fetch/ReferenceSplitQueryTest.java

```diff
@@ -97,196 +100,621 @@ void previousQuerySingleTableInterface() {
-    void keyWithSinglePath() {
+    @DisplayName("Reference on a nullable field")
+    /**
+     * Given that A has a field referencing B and this field is nullable and has a single reference directive with only
+     * the table parameter B, and a direct relation exists from A to B, when a new resolver is generated, then a JOIN
+     * clause should be created. The JOIN clause must include table B retrieved through A.
+     */
+    void nullableField() {
         assertGeneratedContentContains(
-                "keyWithSinglePath", Set.of(CUSTOMER_NOT_GENERATED),
-                ".from(customer_2952383337_address"
+                "nullableField", Set.of(CUSTOMER_NOT_GENERATED),
+//                ".from(customer_2952383337_address"
+                "_customer = CUSTOMER.as",
+                "customer_2952383337_address = _customer.address().as",
+                """
+                .select(
+                        DSL.row(_customer.CUSTOMER_ID),
+                        DSL.row(customer_2952383337_address.getId()).mapping(Functions.nullOnAllNull(Address::new))
+                )
+                .from(_customer)
+                .join(customer_2952383337_address)
+                .where(DSL.row(_customer.CUSTOMER_ID).in(customerResolverKeys.stream().map(Record1::valuesRow).toList()))
+                .fetchMap(Record2::value1, Record2::value2);
+                """
         );
     }
 
     @Test
-    @DisplayName("Key path with multiple possible paths between the tables")
-    void keyWithMultiplePaths() {
+    @DisplayName("Indirect table path")
+    /**
+     * Given that A has a field referencing C and this field has a single reference directive with only the table
+     * parameter B, and there is no direct relation from A to C, but there are relations from A to B and from B to C,
+     * when a new resolver is generated, multiple JOIN clauses should be created. The first JOIN clause must include
+     * table B retrieved through A, and the second JOIN clause must include C retrieved through B.
+     */
+    void throughTable() {
         assertGeneratedContentContains(
-                "keyWithMultiplePaths",
-                ".from(film_3747728953_filmoriginallanguageidfkey"
+                "throughTable", Set.of(CUSTOMER_NOT_GENERATED),
+//                ".from(customer_2952383337_address",
+//                ".join(address_1214171484_city"
+                "_customer = CUSTOMER.as",
+                "customer_2952383337_address = _customer.address().as",
+                "address_1214171484_city = customer_2952383337_address.city().as",
+                """
+               .select(
+                        DSL.row(_customer.CUSTOMER_ID),
+                        DSL.row(address_1214171484_city.getId()).mapping(Functions.nullOnAllNull(City::new))
+                )
+                .from(_customer)
+                .join(customer_2952383337_address)
+                .join(address_1214171484_city)
```

Comment:
Igjen så har vi et felt som er nullable så trenger kanskje noe sånt 🤔 
```suggestion
                .from(_customer)
                .leftJoin(customer_2952383337_address)
                .leftJoin(address_1214171484_city)
```
Men som jeg nevner i kommentaren over så ville jeg ventet til vi får testet i graphitron-example før du endrer på dette! (i tilfelle dette ikke er et problem likevel)